tools/src/tester/helpers.ts

```diff
@@ -0,0 +1,9 @@
+import { type Evaluation, Result } from './types/eval.types'
+
+export function overall_result (evaluations: Evaluation[]): Result {
+  if (evaluations.length === 0) return Result.PASSED
+  if (evaluations.some(e => e.result === Result.ERROR)) return Result.ERROR
+  if (evaluations.some(e => e.result === Result.FAILED)) return Result.FAILED
+  if (evaluations.every(e => e.result === Result.SKIPPED)) return Result.SKIPPED
+  return Result.PASSED
```

Comment:
Those 2 code blocks are not logically equivalent though. In the former, a `Result.ERROR` anywhere in the array will result in `Result.ERROR` being returned. In the latter, if a FAILED appears before an ERROR, FAILED will be returned instead. We can definite incorporate this logic into that switch statement with a flag but it will make it even more complicated. I think trading a bit of performance for a better readability is fair here.

Reply:
That's a fair point regarding the difference in behaviour regarding errors vs failed. You could split the difference between lean more towards your preferred style but still reduce the amount of looping by counting upfront:
```
  const counts = evaluations.map(e => e.result)
    .reduce<Record<string, number>>((c, r) => ({ ...c, [r]: (c[r] ?? 0) + 1 }), { })

  if (counts[Result.ERROR] > 0) return Result.ERROR
  if (counts[Result.FAILED] > 0) return Result.FAILED
  if (counts[Result.SKIPPED] === evaluations.length) return Result.SKIPPED
  return Result.PASSED
```